libvisual/libvisual/lv_defines.h

```diff
@@ -115,8 +115,19 @@
 #define LV_PLUGIN_EXPORT LV_DLL_EXPORT
 
 /* Utility macros */
+
 #ifdef _MSC_VER
   #define __PRETTY_FUNCTION__ __FUNCTION__
 #endif
 
+#if __cplusplus >= 201703L
```

Comment:
@kaixiong could it be that this needs addition of `defined(__cplusplus) && ` ?

Reply:
@kaixiong let me first spell out the questions that I think we need to answer and then try to give a potential answer. For the questions:
1. Is this code used with a C compiler or only C++? Does need it to work from plain C?
2. Can we safely compare an undefined macro against an integer value?
3. Does the `__cplusplus` value reflect reality (more below)?

Potential answers:
1. Seems like yes, we need to support both C and C++ here.
2. It is defined according to https://stackoverflow.com/questions/5085392/what-is-the-value-of-an-undefined-constant-used-in-if so it seems to become a question of style and/or clarity.
3. MSVC [is said](https://bloomfield.online/posts/report-__cplusplus-correctly-with-cmake-and-visual-studio/) to not have reality in `__cplusplus`. The post lays out some options. [Another one](https://developercommunity.visualstudio.com/t/msvc-incorrectly-defines-cplusplus/139261). I think we do want to support MSVC?

PS: https://sourceforge.net/p/predef/wiki/Standards/ does not seem to have any additional insights.

What do you think?

Reply:
@hartwork Looks like Question 1 and 2 are effectively the same question as the C/C++ preprocessor both default to 0 for undefined constants (apart from the handling of `true` and `false`).

Question 3 is addressed in #269 but does not ultimately matter for this PR because the code falls back to compiler extensions (`_Check_return_`) immediately below this.